packages/knip/src/reporters/codeclimate.ts

```diff
@@ -105,12 +106,26 @@ function createLocation(filePath: string, line?: number, col?: number): codeclim
   };
 }
 
-function createFingerprint(filePath: string, message: string, pos?: number): string {
+function createFingerprint(filePath: string, message: string, hashes: Set<string>): string {
   const md5 = createHash('md5');
 
-  md5.update(filePath);
+  md5.update(toRelative(filePath));
   md5.update(message);
-  md5.update(pos?.toString() ?? '');
+
+  // Create copy of hash since md5.digest() will finalize it, not allowing us to .update() again
+  let md5Tmp = md5.copy();
+  let hash = md5Tmp.digest('hex');
+
+  while (hashes.has(hash)) {
+    // Hash collision. This happens if we encounter the same ESLint message in one file
+    // multiple times. Keep generating new hashes until we get a unique one.
+    md5.update(hash);
+
+    md5Tmp = md5.copy();
+    hash = md5Tmp.digest('hex');
+  }
+
+  hashes.add(hash);
```

Comment:
This should definitely be removed if you’re not worried about duplicate messages.

Reply:
Yeah, don't think we can have duplicate `symbol` instances within the same file. Even less so after we'd have implemented https://github.com/webpro-nl/knip/pull/970#issuecomment-2721108208 (cc @atheck).

Reply:
I believe you have the better knowledge on this subject :)

I imagined maybe if the user adds duplicate exports:

```js
export function unused() {}
export function unused() {}
```

or even TypeScript overloads:

```
export declare function add(a: 1, b: 1): 2
export declare function add(a: 2, b: 2): 4
export declare function add(a: number, b: number): number
export function add(a: number, b: number): number {
  return a + b
}
```

This is an edge case though, and not super relevant. I’ll remove this logic.

Reply:
If such an export would be unused a single issue would be reported, same for unused dependencies in `package.json` or unlisted dependencies in a file. Members of enums and classes should be prefixed with the name of the class/enum. Same for unused exports attached to a namespace, etc.

Thanks!